src/MassTransit/Configuration/Registration/MultiBusInstance.cs

```diff
@@ -20,7 +27,40 @@ public MultiBusInstance(TBus bus, IBusInstance instance)
         public IBus Bus => BusInstance;
         public IBusControl BusControl => _instance.BusControl;
         public IHostConfiguration HostConfiguration => _instance.HostConfiguration;
-
         public TBus BusInstance { get; }
+
+        public void Connect(IBusAttachment attachment)
+        {
+            _attachments.Add(attachment);
+        }
+
+        public async Task Start(CancellationToken cancellationToken)
+        {
+            if (HostConfiguration != null)
+                LogContext.SetCurrentIfNull(HostConfiguration.LogContext);
+
+            LogContext.Info?.Log("Starting bus: {Type} instance", InstanceType.Name);
+            await BusControl.StartAsync(cancellationToken).ConfigureAwait(false);
+
+            if (!_attachments.Any())
+                return;
+
```

Comment:
We should wait until the bus is healthy before starting attachments. I think the endpoint observers help here, but the bus health is there to check. We don't want to start Kafka if the broker isn't connected.

Reply:
I'd better go with observer, it seems cleaner for me, IMHO